### ISSUE TITLE
Keras low-precision YAT conv and embedding can return negative or infinite scores

### ISSUE THREAD

**Issue by mlnomadpy** [azettaai/nmn#55]
## Summary

Keras YAT convolution and embedding paths do not clamp cancellation-induced negative squared distances. Under the JAX backend, exact query/kernel matches in float16 and bfloat16 produce infinity or negative YAT scores.

## Affected code

- `src/nmn/keras/conv.py` (all forward and transpose convolution variants)
- `src/nmn/keras/embed.py`

## Reproduction

On commit `4f0298d`, Keras 3.14.1 with JAX 0.9.2:

```python
import jax
import jax.numpy as jnp
from nmn.keras import YatEmbed, YatConv1D

for dtype_name, dtype, seed in [
    ("float16", jnp.float16, 0),
    ("bfloat16", jnp.bfloat16, 8),
]:
    q = jax.random.normal(jax.random.key(seed), (1, 3), dtype=dtype)

    embed = YatEmbed(
        1, 3, use_alpha=False, epsilon=1e-5,
        dtype=dtype_name, embedding_initializer="zeros",
    )
    embed(jnp.array([0]))
    embed.embedding.assign(q)
    print(dtype_name, embed.attend(q))

    x = q.reshape(1, 3, 1)
    kernel = q.reshape(3, 1, 1)
    conv = YatConv1D(
        1, 3, use_bias=False, use_alpha=False,
        epsilon=1e-5, dtype=dtype_name,
        kernel_initializer="zeros",
    )
    conv(x)
    conv.kernel.assign(kernel)
    print(dtype_name, conv(x))
```

Observed for both embedding and convolution:

```
float16:  inf
bfloat16: -69
```

The bfloat16 computed distance is `-0.015625`.

## Impact

Supported Keras dtype policies can silently return invalid YAT values in convolution and tied-embedding output paths.

## Acceptance criteria

- [ ] Clamp distances to zero before epsilon for embedding and all six conv/transpose variants.
- [ ] JAX-backed float16/bfloat16 exact-match tests produce finite, non-negative outputs and gradients.
- [ ] Equivalent TensorFlow-backend tests cover the same dtype policies where supported.
- [ ] Low-precision outputs are compared with an fp32 reference on non-collision inputs.
